translate/convert/l20n2po.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+
+"""Convert Mozilla .l20n files to Gettext PO localization files.
+
+See: http://docs.translatehouse.org/projects/translate-toolkit/en/latest/commands/l20n2po.html
+for examples and usage instructions.
+"""
+
+from translate.storage import po, l20n
+
+
+class l20n2po:
+    """convert a .l20n file to a .po file for handling the
+    translation."""
+
+    def __init__(self, blankmsgstr=False, duplicatestyle="msgctxt"):
+        pass
+
+    def convertl20nunit(self, store, unit):
+        po_units = []
+
+        po_unit = po.pounit(encoding="UTF-8")
+        po_unit.setid(unit.getid())
+        po_unit.addlocation(unit.getid())
+        po_unit.source = unit.value
+        po_units.append(po_unit)
+
+        return po_units
+
+    def convertstore(self, l20n_store):
+        """converts a .l20n file to a .po file..."""
+        target_store = po.pofile()
+        targetheader = target_store.header()
+        targetheader.addnote("extracted from %s" % l20n_store.filename,
+                             "developer")
+        l20n_store.makeindex()
+        for l20nunit in l20n_store.units:
+            pounits = self.convertl20nunit(l20n_store, l20nunit)
+            for pounit in pounits:
+                target_store.addunit(pounit)
+        return target_store
+
+    def mergestore(self, origl20nfile, translatedl20nfile):
+        """converts two .l20n files to a .po file..."""
+        target_store = po.pofile()
+        targetheader = target_store.header()
+        targetheader.addnote("extracted from %s, %s" % (origl20nfile.filename,
+                                                        translatedl20nfile.filename),
+                             "developer")
+        translatedl20nfile.makeindex()
+        for l20nunit in origl20nfile.units:
+            pounits = self.convertl20nunit(origl20nfile, l20nunit)
+            for pounit in pounits:
+                target_store.addunit(pounit)
+                for location in pounit.getlocations():
+                    # TODO: check if using of `id_index` fits here in all cases
+                    if location in translatedl20nfile.id_index:
+                        pounit.target = translatedl20nfile.id_index[location].target
+        return target_store
+
+
+def convertl20n(inputfile, outputfile, templatefile,
+                pot=False, duplicatestyle="msgctxt"):
+    inputstore = l20n.l20nfile(inputfile)
+    convertor = l20n2po(blankmsgstr=pot,
+                        duplicatestyle=duplicatestyle)
+    if templatefile is None:
+        outputstore = convertor.convertstore(inputstore)
+    else:
+        templatestore = l20n.l20nfile(templatefile)
+        outputstore = convertor.mergestore(templatestore, inputstore)
+    if outputstore.isempty():
+        return 0
+    outputfile.write(str(outputstore))
+    return 1
+
+
+formats = {
+    "l20n": ("po", convertl20n),
+    ("l20n", "l20n"): ("po", convertl20n),
```

Comment:
You also need the `ftl` suffix to handle the latest files.  Same would apply to `po2l20n`